pkg/log/checkpoint/files_test.go

```diff
@@ -0,0 +1,128 @@
+package checkpoint
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"runtime"
+	"sort"
+	"testing"
+
+	"github.com/kolide/kit/ulid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFilesFound(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := os.MkdirTemp("", "log-checkpoint-files-test")
+	require.NoError(t, err, "making temp dir")
+
+	var tests = []struct {
+		dirsToCreate int
+		filesPerDir  int
+	}{
+		{dirsToCreate: 2, filesPerDir: 2},
+		{dirsToCreate: 0, filesPerDir: 0},
+	}
+
+	for _, tt := range tests {
+		t.Run("testFilesFound", func(t *testing.T) {
+			dirs, expectedPaths, err := createTestFiles(tempDir, tt.dirsToCreate, tt.filesPerDir)
+			require.NoError(t, err, "creating test files")
+
+			foundPaths := fileNamesInDirs(dirs...)
+			sort.Strings(foundPaths)
+			require.Equal(t, expectedPaths, foundPaths)
+			require.Equal(t, tt.dirsToCreate*tt.filesPerDir, len(foundPaths))
+		})
+	}
+
+	require.NoError(t, os.RemoveAll(tempDir), "deleting temp dir")
+}
+
+func TestDirNotFound(t *testing.T) {
```

Comment:
Is this the same case as `dirsToCreate: 0` above?

Reply:
no, the test case where `dirsToCreate:0` is passing any empty array to the `fileNamesInDirs` function (it's not searching any dirs) where as this one is looking for a directory that does not exist. left a comment and updated variable name for clarity